### ISSUE TITLE
NullPointerException in AbstractMeta.copyFK for SQLite when a foreign key references a unique key

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#11287]
The fixes introduced with #11258 have introduced a regression:

```
[ERROR] Error while initialising type
java.lang.NullPointerException
    at org.jooq.impl.AbstractMeta.copyFK (AbstractMeta.java:664)
    at org.jooq.impl.Snapshot$SnapshotTable.resolveReferences (Snapshot.java:253)
    at org.jooq.impl.Snapshot$SnapshotSchema.resolveReferences (Snapshot.java:152)
    at org.jooq.impl.Snapshot$SnapshotCatalog.resolveReferences (Snapshot.java:115)
    at org.jooq.impl.Snapshot.resolveReferences (Snapshot.java:87)
    at org.jooq.impl.Snapshot.<init> (Snapshot.java:82)
    at org.jooq.impl.AbstractMeta.snapshot (AbstractMeta.java:571)
    at org.jooq.meta.sqlite.SQLiteDatabase.snapshot (SQLiteDatabase.java:120)
    at org.jooq.meta.sqlite.SQLiteTableDefinition.interpretedTable (SQLiteTableDefinition.java:95)
    at org.jooq.meta.sqlite.SQLiteTableDefinition.getElements0 (SQLiteTableDefinition.java:137)
    at org.jooq.meta.AbstractElementContainerDefinition.getElements (AbstractElementContainerDefinition.java:88)
    at org.jooq.meta.AbstractTableDefinition.getColumns (AbstractTableDefinition.java:186)
    at org.jooq.codegen.JavaGenerator.generateTable (JavaGenerator.java:5187)
    at org.jooq.codegen.JavaGenerator.generateTables (JavaGenerator.java:5174)
    at org.jooq.codegen.JavaGenerator.generate (JavaGenerator.java:615)
    at org.jooq.codegen.JavaGenerator.generate (JavaGenerator.java:570)
    at org.jooq.codegen.JavaGenerator.generate (JavaGenerator.java:472)
    at org.jooq.codegen.GenerationTool.run0 (GenerationTool.java:869)
    at org.jooq.codegen.GenerationTool.run (GenerationTool.java:233)
    at org.jooq.codegen.GenerationTool.generate (GenerationTool.java:228)
```

The problem can be reproduced like this:

```java
ctx.execute("create table t1 (i int unique)");
ctx.execute("create table t2 (j int references t1 (i))");
System.out.println(ctx.meta().snapshot());
```

**Comment by lukaseder** [comment]
The underlying problem here is that we don't have any unique key meta information in `MetaImpl` (https://github.com/jOOQ/jOOQ/issues/10823). So, the issue might affect a few other dialects, too.

**Comment by lukaseder** [comment]
The regression isn't visible in the code generator so prominently. It was triggered by an attempt to implement https://github.com/jOOQ/jOOQ/issues/11286.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.15.0 and 3.14.6 (#11288)

**Comment by lukaseder** [comment]
The output is still incorrect:

```sql
create table t1 (
  i int
);
create table t2 (
  j int
);
```

I.e. both the `UNIQUE` constraint, and by consequence, the `FOREIGN KEY` constraint are missing. But at least this doesn't affect any other tables. The missing `UK` will be fixed via https://github.com/jOOQ/jOOQ/issues/10823
